src/miner/error.rs

```diff
@@ -0,0 +1,22 @@
+#[derive(Debug)]
+pub enum MinerError {
```

Comment:
We could remove this entirely if we make `Worker::new` infallible. It's also more correct, because an error can never be returned from that function.